instrumentation/influxdb-2.4/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/influxdb/v2_4/InfluxDbClientTest.java

```diff
@@ -0,0 +1,311 @@
+            databaseName);
+    influxDb.query(
+        query,
+        10,
+        (cancellable, queryResult) -> {
+          countDownLatch.countDown();
+        },
+        () -> {},
+        throwable -> {
+          countDownLatchFailure.countDown();
+        });
+    Assertions.assertFalse(countDownLatchFailure.await(1, TimeUnit.SECONDS));
+    Assertions.assertTrue(countDownLatch.await(1, TimeUnit.SECONDS));
+
+    testing.waitAndAssertTraces(
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span ->
+                    span.hasName("SELECT " + databaseName)
+                        .hasKind(SpanKind.CLIENT)
+                        .hasAttributesSatisfying(
+                            attributeAssertions(
+                                "SELECT MEAN(water_level) FROM h2o_feet where time = ?; SELECT water_level FROM h2o_feet LIMIT ?",
+                                "SELECT",
+                                databaseName))));
+  }
+
+  @Test
+  void testWriteWithFourArguments() {
+    String measurement = "cpu_load";
+    List<String> records = new ArrayList<>();
+    records.add(measurement + ",atag=test1 idle=100,usertime=10,system=1 1485273600");
+    influxDb.write(databaseName, "autogen", InfluxDB.ConsistencyLevel.ONE, records);
+
+    testing.waitAndAssertTraces(
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span ->
+                    span.hasName("write " + databaseName)
+                        .hasKind(SpanKind.CLIENT)
+                        .hasAttributesSatisfying(
+                            attributeAssertions("write", "write", databaseName))));
+  }
+
+  @Test
+  void testWriteWithFiveArguments() {
+    String measurement = "cpu_load";
+    List<String> records = new ArrayList<>();
+    records.add(measurement + ",atag=test1 idle=100,usertime=10,system=1 1485273600");
+    influxDb.write(
+        databaseName, "autogen", InfluxDB.ConsistencyLevel.ONE, TimeUnit.SECONDS, records);
+
+    testing.waitAndAssertTraces(
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span ->
+                    span.hasName("write " + databaseName)
+                        .hasKind(SpanKind.CLIENT)
+                        .hasAttributesSatisfying(
+                            attributeAssertions("write", "write", databaseName))));
+  }
+
+  @Test
+  void testWriteWithUdp() {
+    List<String> lineProtocols = new ArrayList<String>();
+    for (int i = 0; i < 2000; i++) {
+      Point point = Point.measurement("udp_single_poit").addField("v", i).build();
+      lineProtocols.add(point.lineProtocol());
+    }
+    influxDb.write(port, lineProtocols);
+
+    testing.waitAndAssertTraces(
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span ->
+                    span.hasName("write unknown")
```

Comment:
The naming rules for database client span names in https://github.com/open-telemetry/semantic-conventions/blob/main/docs/database/database-spans.md do not suggest using `unknown` when database name is not known, they suggest to omit the database name instead.

Reply:
> The naming rules for database client span names in https://github.com/open-telemetry/semantic-conventions/blob/main/docs/database/database-spans.md do not suggest using `unknown` when database name is not known, they suggest to omit the database name instead.

Thank you for your reminder, I modified it now. But I found in https://github.com/open-telemetry/semantic-conventions/blob/main/docs/database/database-spans.md, it is not clearly stated that it is recommended to omit the database name in this case. Is it necessary to add this situation in the document?

Reply:
I reread this part in the spec and noticed that it was changed a couple of days ago. Now the spec suggests a prioritized [list](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/database/database-spans.md#target-placeholder) of attributes that could be added to the span name. We have not implemented this yet.